### PR TITLE
Public worker api so this module is useful for tests too

### PR DESCRIPTION
**Pull request by lightsofapollo**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
@russnicoletti @garndt

File: examples/work_single_task.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/**
+Example script showing the setup of the worker and posting a single task to show
+en entire work flow for the worker.
```
Comment (russnicoletti) [inline examples/work_single_task.js:3]: s/en/an

File: worker.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          ).then(function(result) {
+            debug('*** Task is reported as completed, status object: ');
+            debug(result.status);
+            debug('*** End of reportCompleted status object');
+            self.emit('task completed', taskId, runId);
+          });
+        }).catch(function(err) {
+          debug(JSON.stringify(err));
+          var err = new Error('Failed to work task');
+          err.body = err.body;
```
Comment (russnicoletti) [inline worker.js:116]: What does this line do?

File: worker.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+function listenerCredentials(queue) {
+  // Try a cache environment variable/default first.
+  if (process.env.TASKCLUSTER_AMQP_URL) {
+    return Promise.resolve(process.env.TASKCLUSTER_AMQP_URL);
+  }
+
+  // Fallback to the deprecated queue endpoint.
+  return queue.getAMQPConnectionString().then(function(result) {
+    return result.url;
+  });
```
Comment (russnicoletti) [inline worker.js:25]: Should have a 'catch' here?

File: worker.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+*/
+function listenerCredentials(queue) {
+  // Try a cache environment variable/default first.
+  if (process.env.TASKCLUSTER_AMQP_URL) {
+    return Promise.resolve(process.env.TASKCLUSTER_AMQP_URL);
+  }
+
+  // Fallback to the deprecated queue endpoint.
+  return queue.getAMQPConnectionString().then(function(result) {
+    return result.url;
```
Comment (russnicoletti) [inline worker.js:24]: Should this be 'Promise.resolve(result.url)' ?

Comment (lightsofapollo) [inline worker.js:24]: You can return from a promise with the same effect (you can also chain promises) [i.e. this code does what you expect]

File: worker.js (hunk 2 above)
Comment (lightsofapollo) [inline worker.js:116]: oops.. nothing! typo

File: worker.js (hunk 3 above)
Comment (lightsofapollo) [inline worker.js:25]: This intentionally allows the error to "bubble" up from the promise chain in the caller (in the example it handles the catch case)

**Comment by jonasfj** [comment]
No longer relevant closing... as this is now very old.

**Pull request closed without merging**